Modules/SubModule_ASL/xASL_wrp_RealignASL.m

```diff
@@ -139,7 +136,7 @@ function xASL_wrp_RealignASL(x,bSubtraction)
 %     end
 
 % Run motion correction for corresponding case
-if nFrames>2 && bSubtraction && (x.modules.asl.bMultiPLD  ||  x.modules.asl.bMultiTE  || x.modules.asl.bTimeEncoded)
+if nFrames>2 && bSubtraction && (x.modules.asl.bMultiPLD  ||  x.modules.asl.bMultiTE)

```

Comment:
Incorrect, this is a check to state if we can run motion correction at all right, this is irrespective of multiPLD and multiTE...

Also, I don't know exactly what `bSubtraction` does here, this boolean is here to differentiate between `spm_realign` (`bSubtraction==false`) and `spm_realign` with zig-zag (`bSubtraction==true`).
We might replace `bSubtraction` by `bZigZag` for clarification

Reply:
I agree this is unclear, maybe we should add comments for these `if` statements to show what is happening

Reply:
Yes. Unclear. Plus it should have reused bZigZag assigned previously here :D This is now fixed.

`bSubtraction` was there previously, to activate ZigZag. This is now renamed to `bASL` and for non-ASL, `bZigZag` is turned off. And the whole `bZigZag` assignment is done above.

With all the above - makes no sense to rename `bSubtraction` to `bZigZag` as there are cases with `bSubtraction==1` that still have `bZigZag` off. Like if it contains deltaMs - is a subtraction-based technique, but ZigZag has to be off - this is all explained in the comment above and fixed in this commit: 3d240036

Reply:
Well you can say the same for `bASL`, there are also cases where `bASL` is true and `bZigZag` is off.
I would keep it simplier.

We need as options inside `xASL_wrp_Realign`:
`bMotionCorrection`
`bSubtraction` (used previously only to switch between ZigZag off-on) -> `bZigZag`
`bENABLE`

In the header of `xASL_wrp_Realign`, we set these three options based on the type of ASL (`bMultiPLD`, `bContainsDeltaM`, `bMultiTE`, etc)

Reply:
That's not what is described in the header. Header say bSubtraction == bASL == ASL yes/no

Yes, we have now both `bZigZag` and `bASL`. We don't need `bMotionCorrection`, all is motion corrected. We only turn of ZigZag, but not depending on function parameters, but on data parameters. We need to keep this transparent and adjust based on data. To me, it makes no sense to start gluing up extra functionality - that's what you always protest against ;)

Reply:
If you want to change functionality without messing up things with too much backwards compatibility, we can just tackle this in the next major release. For me the current header and descriptions seem pretty clear.